HKMP/Game/Client/PlayerManager.cs

```diff
@@ -37,9 +37,14 @@ internal class PlayerManager {
         public Team LocalPlayerTeam { get; private set; } = Team.None;
 
         /// <summary>
-        /// The player prefab GameObject.
+        /// A collection of pre-instantiated players that will be fetched when spawning a player.
         /// </summary>
-        private readonly GameObject _playerPrefab;
+        private Dictionary<ushort, GameObject> _players = new();
+
+        /// <summary>
+        /// The maximum number of players that may exist on a server at once.
+        /// </summary>
+        private const byte MaxPlayers = byte.MaxValue;
```

Comment:
The maximum number of players on a server is not equal to the max value of a byte. Rather it is equal to the max value of an unsigned short, or `65535`. I'm unsure at what point object pooling is advantageous and when it pre-instantiates unnecessarily many player prefabs.

Reply:
Perhaps a better implementation of this would be to pool a comparatively small amount of player objects (compared to the max player count of 65536) but still enough to cover most server loads (e.g. 255, or maybe it can be configurable), and if that pool were to be exhausted, then additional player objects could be added to the pool. This way the advantages of pooling could still be had when a player enters the same scene as another.